projects/hello_buttons_adcs/src/main.c

```diff
@@ -0,0 +1,58 @@
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+
+GpioAddress adc_pin_addr = { .port = GPIO_PORT_A, .pin = 6 };
+GpioAddress button_addr = { .port = GPIO_PORT_B, .pin = 2 };
```

Comment:
These should be marked `static` and prefixed with `s_` (and could also be `const`).

Reply:
Is there a general rule for using `static` and `const`s? I noticed in some projects like the pedai board `main.c`, the enums for storage were marked `static` and some the settings were marked `const` (but not the I2C settings).

Reply:
In general, anything you don't want to be accessible from other modules should be `static` - the keyword makes it not visible. This includes pretty much any global variable and any private functions. For `const`, in general it's a good idea to mark anything `const` that shouldn't change, like settings. (In pedal board's main.c, the `can_settings` struct really should be `static`.)